cmake/ExternalProjectHelper.cmake

```
@@ -29,7 +29,8 @@ function(nrn_initialize_submodule path)
   endif()
   message(STATUS "Sub-module : missing ${path} : running git submodule update --init")
   execute_process(
-    COMMAND ${GIT_EXECUTABLE} submodule update --init -- ${path}
+    # Shallow clone by default. Developers should clone recursivelly
+    COMMAND ${GIT_EXECUTABLE} submodule update --recursive --init --depth=1 -- ${path}
```

```suggestion
    COMMAND ${GIT_EXECUTABLE} submodule update --recursive --init -- ${path}
```
Unless it's required I'd split this, so it can be discussed separately.

---

Notice this only applies to submodules. Is there any reason we would require the submodules history?

---

I'd spin this off, too. Seems pretty unrelated. And, maybe, git settings could be steered via a more appropriate place? Like this setting: https://git-scm.com/docs/gitmodules#Documentation/gitmodules.txt-submoduleltnamegtshallow